graphql/resolvers.go

```diff
@@ -1237,6 +1237,14 @@ func (r *patchResolver) BaseTaskStatuses(ctx context.Context, obj *restModel.API
 	return getAllTaskStatuses(baseTasks), nil
 }
 
+func (r *patchResolver) TaskStatusCounts(ctx context.Context, obj *restModel.APIPatch, options *BuildVariantOptions) ([]*task.StatusCount, error) {
```

Comment:
I don't think we should do this. Patches don't technically have tasks. The version is what contains the tasks. 

I think we should instead make it possible for a patch to resolve a version. So the caller can resolve the TaskStatusCounts from that resolver instead. 

So the call would look like this. 
```
user {
patches {
  version {
    taskStatusCounts
  }
} 
}

```

Reply:
if the patch is unconfigured and has no version  and no tasks it will just return null.

Reply:
Sounds good! I originally thought to create the separate field to save us a query into the Version collection when querying patches but it seems useful to query any Version field from a patch.